language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableCodeAction.java

```diff
@@ -166,12 +177,53 @@ static class CreateVariableOut {
         List<String> types;
         List<TextEdit> edits;
         List<TextEdit> imports;
+        List<Integer> renamePositions;
 
-        public CreateVariableOut(String name, List<String> types, List<TextEdit> edits, List<TextEdit> imports) {
+        public CreateVariableOut(String name, List<String> types, List<TextEdit> edits, List<TextEdit> imports,
+                                 List<Integer> renamePositions) {
             this.name = name;
             this.types = types;
             this.edits = edits;
             this.imports = imports;
+            this.renamePositions = renamePositions;
+        }
+    }
+
+    public void addRenamePopup(CodeActionContext context, List<TextEdit> textEdits,
```

Comment:
Let's modify this addRenamePopup method to get the variableTextEdit and other additional edits separately so that we don't have to assume that the first text edit is always the text edit corresponding to the variable declaration.